<a href="https://colab.research.google.com/github/AatikaCodes/MSSP6070_001/blob/main/WeeklyModules/Week04/MSSP6070_001Week4Activity_Aatika.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [15]:
import numpy as np
import pandas as pd
import matplotlib as mpl
import matplotlib.pyplot as plt
import statsmodels.formula.api as sm
import statsmodels.api as se
import seaborn as sns
import datetime as dt
from sklearn.linear_model import LinearRegression
from scipy import stats
from io import StringIO
import itertools
import plotly.express as px

pd.set_eng_float_format(accuracy =3, use_eng_prefix=True)
pd.set_option('display.float_format', lambda x: '%.2f' % x)
#pd.set_option('display.max_columns', None)
#pd.set_option('display.max_rows', None)

from google.colab import userdata
import os
github_token = userdata.get('Git_Key')
owner = 'AatikaCodes' # Replace with the GitHub repository owner
repository = 'MSSP6070_001' # Replace with the GitHub repository name
clone_url = f'https://{github_token}@github.com/{owner}/{repository}.git'
# Clone the repository
import subprocess
subprocess.run(['git', 'clone', clone_url])
# Navigate into the cloned repository directory (optional, but often useful)
os.chdir(repository)
print(f"Changed directoGit_HUBry to: {os.getcwd()}")






Changed directoGit_HUBry to: /content/MSSP6070_001/MSSP6070_001


**Ques 1.** Download the Working with Pandas DataFrame files from the Course File/Files/Working with Panda DataFrame folder and upload them to your Google Colab instance.  You will use the 'parsed.csvDownload parsed.csv' file for this exercise.

a. Create a folder structure that allows you to use the Python Notebooks and data files in GitHub.

In [16]:
parsed = pd.read_csv('/content/drive/MyDrive/MSSPDA_6070_Aatika/Modules/Week04/parsed.csv')

In [17]:
parsed

,alert,cdi,code,detail,dmin,felt,gap,ids,mag,magType,...,status,time,title,tsunami,type,types,tz,updated,url,parsed_place
0,NaN,NaN,37389218,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.01,NaN,85.00,",ci37389218,",1.35,ml,...,automatic,1539475168010,"M 1.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.00,1539475395144,https://earthquake.usgs.gov/earthquakes/eventp...,California
1,NaN,NaN,37389202,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.02,NaN,79.00,",ci37389202,",1.29,ml,...,automatic,1539475129610,"M 1.3 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.00,1539475253925,https://earthquake.usgs.gov/earthquakes/eventp...,California
2,NaN,4.40,37389194,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.02,28.00,21.00,",ci37389194,",3.42,ml,...,automatic,1539475062610,"M 3.4 - 8km NE of Aguanga, CA",0,earthquake,",dyfi,focal-mechanism,geoserve,nearby-cities,o...",-480.00,1539536756176,https://earthquake.usgs.gov/earthquakes/eventp...,California
3,NaN,NaN,37389186,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.03,NaN,39.00,",ci37389186,",0.44,ml,...,automatic,1539474978070,"M 0.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.00,1539475196167,https://earthquake.usgs.gov/earthquakes/eventp...,California
4,NaN,NaN,73096941,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.08,NaN,192.00,",nc73096941,",2.16,md,...,automatic,1539474716050,"M 2.2 - 10km NW of Avenal, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.00,1539477547926,https://earthquake.usgs.gov/earthquakes/eventp...,California
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
9327,NaN,NaN,73086771,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.02,NaN,185.00,",nc73086771,",0.62,md,...,reviewed,1537230228060,"M 0.6 - 9km ENE of Mammoth Lakes, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.00,1537285598315,https://earthquake.usgs.gov/earthquakes/eventp...,California
9328,NaN,NaN,38063967,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.03,NaN,50.00,",ci38063967,",1.00,ml,...,reviewed,1537230135130,"M 1.0 - 3km W of Julian, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.00,1537276800970,https://earthquake.usgs.gov/earthquakes/eventp...,California
9329,NaN,NaN,2018261000,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.45,NaN,276.00,",pr2018261000,",2.40,md,...,reviewed,1537229908180,"M 2.4 - 35km NNE of Hatillo, Puerto Rico",0,earthquake,",geoserve,origin,phase-data,",-240.00,1537243777410,https://earthquake.usgs.gov/earthquakes/eventp...,Puerto Rico
9330,NaN,NaN,38063959,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.02,NaN,61.00,",ci38063959,",1.10,ml,...,reviewed,1537229545350,"M 1.1 - 9km NE of Aguanga, CA",0,earthquake,",focal-mechanism,geoserve,nearby-cities,origin...",-480.00,1537230211640,https://earthquake.usgs.gov/earthquakes/eventp...,California


**Ques 2. Complete the following exercise:**

a. Find the 95th percentile of earthquake magnitude in Japan using magType of 'mb'

b. Find the percentage of earthquakes in Indonesia that were coupled with tsunamis.

c. Get summary statistics for earthquakes in Nevada

d. Add a column to the dataframe indicating whether or not the earthquake happened in a country or US state that is on the Ring of Fire. Use Bolivia, Chile, Ecuador, Peru, Costa Rica, Guatemala, Mexico (be sure not to select New Mexico), Japan, Philippines, Indonesia, New Zealand, Antarctica (look for the Antarctic), Canada, Fiji, Alaska, Washington, California, Russia, Taiwan, Tonga, and the Kermadec Islands).

e. Calculate the number of earthquakes in the Ring of Fire locations and the number outside of them,

f. Find the tsunami count along the Ring of Fire.

**Ques. 2.a)** Find the 95th percentile of earthquake magnitude in Japan using **magType of 'mb'**

In [18]:
japan_mb = parsed[(
    parsed['parsed_place'] == 'Japan') &
    (parsed['magType'] == 'mb')
]

float(japan_mb['mag'].quantile(0.95))

4.9

Answer: 95th percentile of earthquake magnitude in Japan using magType of 'mb': 4.9

**Ques.2.b)** Find the percentage of earthquakes in Indonesia that were coupled with tsunamis.

In [19]:
indonesia = parsed[parsed['parsed_place'] == 'Indonesia']

print(f"{indonesia['tsunami'].mean() * 100:.2f}%")

23.13%


Answer: 23.13% of earthquakes in Indonesia are coupled with tsunamis. There are 147 earthquakes in Indonesia, and 34 had tsunamis.

**Ques.2.c)** Get summary statistics for earthquakes in Nevada

In [20]:
nevada = parsed[parsed['parsed_place'] == 'Nevada']

nevada['mag'].describe()

,mag
count,681.00
mean,0.50
std,0.70
min,-0.50
25%,-0.10
50%,0.40
75%,0.90
max,2.90


Answer: The summary statistics shows here the distribution of earthquake magnitudes in Nevada. There were (681.00) earthquakes, with an average magnitude of (0.50). The median magnitude was (0.40), while the magnitudes ranged from (-0.50) to (2.90). The standard deviation of (0.70) indicates the amount of variation in earthquake magnitudes. The 25th percentile was (-0.10), meaning 25% of the earthquakes had magnitudes at or below this value, while the 75th percentile was (0.90), meaning 75% had magnitudes at or below this value.

**Ques. 2.d)** Add a column to the dataframe indicating whether or not the earthquake happened in a country or US state that is on the Ring of Fire. Use Bolivia, Chile, Ecuador, Peru, Costa Rica, Guatemala, Mexico (be sure not to select New Mexico), Japan, Philippines, Indonesia, New Zealand, Antarctica (look for the Antarctic), Canada, Fiji, Alaska, Washington, California, Russia, Taiwan, Tonga, and the Kermadec Islands).

In [21]:
ring_of_fire = [
    'Bolivia', 'Chile', 'Ecuador', 'Peru', 'Costa Rica',
    'Guatemala', 'Mexico', 'Japan', 'Philippines', 'Indonesia',
    'New Zealand', 'Canada', 'Fiji', 'Alaska', 'Washington',
    'California', 'Russia', 'Taiwan', 'Tonga', 'Kermadec Islands'
]

parsed['ring_of_fire'] = (
    parsed['parsed_place'].isin(ring_of_fire) |
    parsed['parsed_place'].str.contains('Antarctic', case=False, na=False)
)

In [22]:
parsed['ring_of_fire'].value_counts() #checking the new columns

,count
ring_of_fire,
True,7188
False,2144


Answer: Ring of Fire: 7,188 earthquakes

Outside Ring of Fire: 2,144 earthquakes

Totaling to 9,332, that matches the number of rows in the CSV.

**Ques. 2.e)** Calculate the number of earthquakes in the Ring of Fire locations and the number outside of them.

In [23]:
parsed['ring_of_fire'].value_counts()

,count
ring_of_fire,
True,7188
False,2144


Answer: The new 'ring_of_fire' column identifies whether each earthquake occurred in a specified Ring of Fire location. The 'True' category represents earthquakes that occurred in a Ring of Fire country, U.S. state, or region, while 'False' represents earthquakes outside those locations. Moreover, the ring_of_fire column identifies whether each earthquake occurred in one of the specified Ring of Fire locations. Of the total 9,332 earthquakes in the dataset, 7,188 (about 77.02%) occurred in Ring of Fire locations, while 2,144 (about 22.98%) occurred outside those locations.

**Ques 2.f)** Find the tsunami count along the Ring of Fire.

In [24]:
float(parsed.loc[parsed['ring_of_fire'], 'tsunami'].sum())

45.0

Answer: There were 45 tsunami-associated earthquakes along the Ring of Fire.